dspace-api/src/main/java/org/dspace/statistics/AnonymizeStatistics.java

```diff
@@ -0,0 +1,273 @@
+            printHelp(options);
+            System.exit(-1);
+        }
+
+        if (commandLine.hasOption(SLEEP_OPTION)) {
+            sleep = parseInt(commandLine.getOptionValue(SLEEP_OPTION));
+        }
+
+        if (commandLine.hasOption(BATCH_SIZE_OPTION)) {
+            batchSize = parseInt(commandLine.getOptionValue(BATCH_SIZE_OPTION));
+        }
+
+        if (commandLine.hasOption(THREADS_OPTION)) {
+            threads = parseInt(commandLine.getOptionValue(THREADS_OPTION));
+        }
+    }
+
+    private static void printHelp(Options options) {
+        new HelpFormatter().printHelp("dsrun " + AnonymizeStatistics.class.getCanonicalName(), options);
+    }
+
+    private static void printInfo(String info) {
+        System.out.println(info);
+        log.info(getHeader(context, action, info));
+    }
+
+    private static void printWarning(String warning) {
+        System.out.println(warning);
+        log.warn(getHeader(context, action, warning));
+    }
+
+    private static void printError(Exception error) {
+        error.printStackTrace();
+        log.error(getHeader(context, action, error.getMessage()), error);
+    }
+
+
+    private static void anonymizeStatistics() {
+        try {
+            long updated = 0;
+            long total = getDocuments().getResults().getNumFound();
+            printInfo(total + " documents to update");
+
+            ExecutorService executorService = Executors.newFixedThreadPool(threads);
+
+            QueryResponse documents;
+            do {
+                documents = getDocuments();
+
+                Collection<Callable<Boolean>> callables = new ArrayList<>();
+                Set<String> shards = new HashSet<>();
+
+                for (SolrDocument document : documents.getResults()) {
+                    updated++;
+
+                    callables.add(new DoProcessing(document, updated));
+                    String shard = (String) document.getFieldValue("[shard]");
+
+                    if (isNotBlank(shard)) {
+                        shards.add(shard);
+                    }
+                }
+
+                executorService.invokeAll(callables);
+
+                solrLoggerService.commit();
+
+                for (String shard : shards) {
+                    solrLoggerService.commitShard(shard);
+                }
+
+                System.out.println("processed " + updated + " records");
+            } while (documents.getResults().getNumFound() > 0);
+
+            printInfo(updated + " documents updated");
+            if (updated == total) {
+                printInfo("all relevant documents were updated");
+            } else {
+                printWarning("not all relevant documents were updated, check the DSpace logs for more details");
+            }
+
+        } catch (Exception e) {
+            printError(e);
+        }
+    }
+
+    private static QueryResponse getDocuments() throws SolrServerException, IOException {
```

Comment:
Could we describe this method with JavaDocs *or* add an inline comment above the query to describe *which* documents it's getting.  It seems like it's returning all docs from the last 90 days (or the configured time period)? Is that correct?